### PR TITLE
Add WebApp.addUpdatedNotifyHook

### PR DESCRIPTION
**Pull request by brucejo75**
Add another hook to the `webapp` package.

### `WebApp.addUpdatedNotifyHook`
This function will add a hook to notify the caller whenever meteor has updated its code.  This hook would mostly be used in development to enable the caller to perform actions whenever meteor has updated itself.

@filipenevola, @StorytellerCZ,
This is a hook that I require to enable `autoupdate` when I am developing with a meteor server shared among multiple client applications.  Super useful during development.

Hoping we can sneak this into 2.4 while I was updating `webapp` anyway.

### CODE REVIEW

**Comment by brucejo75** [comment]
Pushed updates to the documentation: https://github.com/meteor/docs/pull/738#issuecomment-907904225

File: packages/webapp/webapp_server.js [hunk 1]
```diff
@@ -412,17 +415,32 @@ function getBoilerplateAsync(request, arch) {
   }));
 }
 
+// Notification hook whenever the runtime configuration is updated
+// hook will pass an object with the following fields:
+// - arch
+// - manifest
+// - runtimeConfig
+WebApp.addUpdatedNotifyHook = function(hook) {
```
Comment (StorytellerCZ) [inline packages/webapp/webapp_server.js:423]: We have the `callback-hook` package which could have been used here.

**Review by StorytellerCZ — COMMENTED** [review]
Missing an entry in history file.

File: packages/webapp/webapp_server.js (hunk 1 above)
Comment (StorytellerCZ) [reply packages/webapp/webapp_server.js:423]: We still have some time before release, so that would be great! Just make sure that it works as you desired.

Comment (brucejo75) [reply packages/webapp/webapp_server.js:423]: done: https://github.com/meteor/meteor/pull/11615

**Pull request merged**